etc/dbus-serialbattery/MNB_BMS.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+from battery import Protection, Battery, Cell
+from struct import *
+from utilis_max17853 import *
+
+class MNB_Protection(Protection):
+
+    def __init__(self):
+        super(MNB_Protection, self).__init__()
+        self.voltage_high_cell = False
+        self.voltage_low_cell = False
+        self.short = False
+        self.IC_inspection = False
+        self.software_lock = False
+
+    def set_voltage_high_cell(self, value):
+        self.voltage_high_cell = value
+        self.set_cell_imbalance(2 if self.voltage_low_cell \
+                                        or self.voltage_high_cell else 0)
+
+    def set_voltage_low_cell(self, value):
+        self.voltage_low_cell = value
+        self.set_cell_imbalance(2 if self.voltage_low_cell \
+                                        or self.voltage_high_cell else 0)
+
+    def set_short(self, value):
+        self.short = value
+        self.set_cell_imbalance(2 if self.short \
+                                        or self.IC_inspection \
+                                        or self.software_lock else 0)
+
+    def set_ic_inspection(self, value):
+        self.IC_inspection = value
+        self.set_cell_imbalance(2 if self.short \
+                                        or self.IC_inspection \
+                                        or self.software_lock else 0)
+
+    def set_software_lock(self, value):
+        self.software_lock = value
+        self.set_cell_imbalance(2 if self.short \
+                                        or self.IC_inspection \
+                                        or self.software_lock else 0)
+
+class MNB(Battery):
+
+    def __init__(self, port,baud,address=0):
+        super(MNB, self).__init__(port,baud)
+        self.hardware_version = 1.02
+        self.voltage = 26
+        self.charger_connected = None
+        self.load_connected = None
+        self.command_address = address
+        self.cell_min_voltage = None
+        self.cell_max_voltage = None
+        self.cell_min_no = None
+        self.cell_max_no = None
+        self.poll_interval = 2000
+        self.type = self.BATTERYTYPE
+        self.capacity = None
+        self.capacity_remain = None
+        self.current = None
+        self.temp3 = None
+        self.temp4 = None
+        self.cells_v= [3.3,3.31,3.33,3.34,3.35,3.36,3.37]
```

Comment:
Convert these cells_v and cells_b to the cell object to link to the rest of the driver